tests/acceptance/16_cf-serverd/keep_promises_fail.cf

```diff
@@ -0,0 +1,43 @@
+body common control
+{
+  inputs => { "../default.cf.sub" };
+  bundlesequence => { default("$(this.promise_filename)") };
+}
+
+bundle agent test
+{
+  commands:
+    "/bin/echo -n '=line_number=' && /bin/grep -n keep_promises_fail.cf $(this.promise_dirname)/../../../cf-serverd/server_transform.c | cut -d: -f1" 
+      contain => shell_command,
+      module => "true";
+
+    "/bin/echo -e '
+      b cf-serverd/server_transform.c:$(echo.line_number)
+      commands
+        set classes_acl=0
+        continue
+      end
+      run
+      quit $_exitcode' > $(sys.workdir)/keep_promises_fail.gdb"
+      contain => shell_command;
+
+    "$(this.promise_dirname)/../../../libtool --mode=execute gdb -x $(sys.workdir)/keep_promises_fail.gdb --args $(this.promise_dirname)/../../../cf-serverd/cf-serverd --no-fork --file $(this.promise_dirname)/keep_promises_fail.cf.sub"
+      classes => keptcode;
+}
```

Comment:
@vpodzime something like the above? It seemed to work pretty well. I will work on the other two instances when I can.

Reply:
Yeah, something like that.